### ISSUE TITLE
Support Microsoft.Extensions.Logging

### ISSUE THREAD

**Issue by alhardy** [AppMetrics/AppMetrics#189]
Dependency on MS logging has been removed and replaced with LibLog. Provide support for app metrics log output to MS logging.

**Comment by johnduhart** [comment]
@alhardy Should this be implemented in LibLog itself?

**Comment by alhardy** [comment]
Yep that would be nice, I did take a very quick look at implementing that, didn't have the time to implement it

**Comment by damianh** [comment]
Hi,

I've did not design LibLog to target other logging abstractions (`Microsoft.Extensions.Logging`, `Common.Logging`) etc. It's purpose is to hook into the logging framework used by the the hosting _application_  with minimal wireup and mitigate transient package dependency issues.

And then Microsoft came along with their conforming logging abstraction (like their [DI container](https://simpleinjector.org/blog/2016/06/whats-wrong-with-the-asp-net-core-di-abstraction/)).

My usual guidance would be:

 - Connect Microsoft.Extensions.Logging to log provider of choice (serilog etc)
 - LibLog enabled libs connect to log provider of choice (serilog etc)

What would the advantage be for LibLog to go to Microsoft.Extensions.Logging ?

Nice project btw ;)

Cheers,
Damian

**Comment by alhardy** [comment]
Cheers @damianh , yep that's been my approach.

Thanks for Liblog, love the approach ;)

**Comment by ugumba** [comment]
My project uses Microsoft.Extensions.Logging (as is normal with ASP.NET Core), configured with Serilog for formatting and sinks.  
I'm unable to get App.Metrics logs.  I don't think forcing apps to adopt LibLog is an acceptable solution.

**Comment by damianh** [comment]
Libraries using LibLog don't force their consumers to **adopt** LibLog. LibLog allows a library to provide it's own logging interface without taking on a dependency and the version management that comes with that. The LibLog autowire up is icing.

However if a library was strongly coupled to AspNetCore, i.e. is middleware, then my advice would be to use the logging abstraction that comes along with that. For example, my ProxyKit project doesn't use LibLog for this reason. Also, it's not exactly fun supporting multiple AspNetCore / MS.Extensions versions.

Ultimately It's up to @alhardy to decide if this project is strongly couple to AspNetCore or not...

**Comment by damianh** [comment]
Also if there is a bug in things not working, I'd like to hear about it.

**Comment by ugumba** [comment]
Thanks for the input - I understand the problem and the rationale.
In my case, it's tricky to get App.Metrics logs into my service's regular configuration, as LibLog detects Serilog's presence and hooks onto the static Serilog.Logger.  For my service, Serilog.Logger is not set, as I require all loggers to be created via DI (MS ILogger, with Serilog for formatting and sinks).  
I'll get by somehow.

**Comment by damianh** [comment]
@ugumba your comment has made me reflect on the role and future of LibLog - I appreciate that. After thinking about it quite a bit I've decided to [obsolete the library](https://github.com/damianh/LibLog/issues/270).

tl;dr: WIth .NET Core 3.1 LTS around the corner and `Microsoft.Extensions.Logging.Abstractions` effectively being shipped as part of platform, it is now the defacto interface (along with all the other items such as DI). I assert that library developers should migrate to it as part of their ongoing .NET Core support. Library developers who still want to support .NET Framework (even via NetStandard) will have some considerations to make.

@alhardy You might want to re-open this. 

Cheers.

**Comment by ugumba** [comment]
Thanks for the update, @damianh - I think it's a correct and wise decision.

**Comment by jonfreedman** [comment]
Does this need to be reopened? It's currently labelled "wontfix" but seems like it should be fixed.

**Comment by ugumba** [comment]
@jonfreedman, it's unlikely that anyone will pick up issues in this repo.  Good as it was, I'd recommend moving on to System.Diagnostics.Metrics.
